src/java.base/share/classes/sun/security/ssl/SSLConfiguration.java

```diff
@@ -133,6 +139,25 @@ final class SSLConfiguration implements Cloneable {
         useExtendedMasterSecret = supportExtendedMasterSecret;
     }
 
+    /*
+     * If the jdk.tls.maxClientCertificateChainLength or jdk.tls.maxServerCertificateChainLength
+     * system property is not set, it will default to the value of the
+     * jdk.tls.maxCertificateChainLength system property. If either of them is
+     * set, it will supersede the value of the jdk.tls.maxCertificateChainLength
+     * system property.
+     */
+    static {
+        Integer clientLen = GetIntegerAction.privilegedGetProperty(
```

Comment:
I think you could call `privilegedGetProperty` with the default value as second argument.

Reply:
Done.

Reply:
I did update the code with his suggestion, and please see it in webrev **01: Incremental.**  However, I further change the code based on the updated CSR, that the new client and server properties have different default values now  from the existing property, and the need to know whether the new property is explicitly set or not. We want to make sure the new properties are not overridden when they are set.